mParticle-Apple-SDK/Data Model/MPConsumerInfo.mm

```diff
@@ -176,39 +176,54 @@ - (NSDictionary *)dictionaryRepresentation {
 @end
 
 #pragma mark - MPConsumerInfo
+@interface MPConsumerInfo ()
+@property (nonatomic, strong, nonnull) dispatch_queue_t queue;
+@end
+
 @implementation MPConsumerInfo
 
 @synthesize cookies = _cookies;
 @synthesize mpId = _mpId;
 @synthesize uniqueIdentifier = _uniqueIdentifier;
 
 - (id)init {
+    // Serial since read/write lock not appropriate given how mpId is created.
+    dispatch_queue_t newQueue = dispatch_queue_create("com.mParticle.MPCookie.queue", DISPATCH_QUEUE_SERIAL);
+    
     self = [super init];
-    if (self) {
+    if (self && newQueue) {
```

Comment:
Thank you for the suggestion and explanation @BrianDoig, `dispatch_once` is indeed appropriate for this case. We do utilize it in other parts of the SDK and also in the implementation and initialization of kits.